src/ethereum/frontier/vm/memory.py

```diff
@@ -0,0 +1,112 @@
+"""
+Ethereum Virtual Machine (EVM) Memory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+EVM memory operations.
+"""
+from ethereum.frontier.vm import Evm
+from ethereum.utils import ceil32
+
+from ...base_types import U256, Bytes
+from .gas import calculate_memory_gas_cost, subtract_gas
+
+
+def extend_memory_and_subtract_gas(
+    evm: Evm, start_position: U256, size: U256
+) -> None:
+    """
+    Extends the size of the memory and
+    substracts the gas amount to extend memory.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    start_position :
+        Starting pointer to the memory.
+    size :
+        Amount of bytes by which the memory needs to be extended.
+    """
+    memory = evm.memory
+    memory_size = U256(len(memory))
+    before_size = ceil32(memory_size)
+    after_size = ceil32(start_position + size)
+    already_paid = calculate_memory_gas_cost(before_size)
+    total_cost = calculate_memory_gas_cost(after_size)
+    if total_cost < already_paid:
+        return None
+    to_be_paid = total_cost - already_paid
+    evm.gas_left = subtract_gas(evm.gas_left, to_be_paid)
+    memory_extend(evm.memory, start_position, size)
+
+
+def memory_extend(memory: bytearray, start_position: U256, size: U256) -> None:
+    """
+    Extends size of memory
+
+    Parameters
+    ----------
+    memory :
+        Memory contents of the EVM.
+    start_position :
+        Starting pointer to the memory.
+    size :
+        Amount by which the memory needs to be extended.
+    """
+    if size == 0:
+        return None
+    memory_size = U256(len(memory))
+    new_size = ceil32(start_position + size)
```

Comment:
```suggestion
    current_memory_size = Uint(len(memory))
    expanded_memory_size = ceil32(start_position + size)
```
I feel these variables add more readability. Theoretically, the memory size is not limited and hence we should be using `Uint` instead of `U256`. Same goes for the output of ceil32 in above.

Reply:
I have used `before_size` and `after_size` which IMO are quite similar to this suggestion here. However, @SamWilsn I'll commit this if you feel strongly about it.